File: 栈的压入、弹出序列.py
# -*- coding:utf-8 -*-
##运行时间：47ms  占用内存：5624k
class Solution:
    def IsPopOrder(self, pushV, popV):
        # write code here
        stack=[]
        while popV:
            if stack and stack[-1]==popV[0]:
                stack.pop()
                popV.pop(0)
            elif pushV:
                stack.append(pushV.pop(0))
            else:
                return False
        return True    
                

File: 栈的压入、弹出序列.cpp
//运行时间：4ms  占用内存：476k
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        if(pushV.size()==0) return false;
        stack<int> s;
        int i,j;
        j=0;
        for(i=0;i<pushV.size();i++)
        {
         s.push(pushV[i]);
         while(j<popV.size() && s.top()==popV[j]){
          s.pop();
          j++;
         }
        }
        return s.empty();
    }
};